Lib/test/test_pty.py

```diff
@@ -76,6 +76,17 @@ def expectedFailureIfStdinIsTTY(fun):
         pass
     return fun
 
+
+def write_all(fd, data):
+    while True:
```

Comment:
Why loop?

Reply:
Oh. My first implementation (before I created this PR) actually implemented write_all(): retry os.write() until all bytes are written. But I was told that it must not happend, so I changed my PR to raise an exception instead.

I will remove the loop until buildbots complete, before merging this change.

At least, with this change, if the test hangs again, we will know for sure that no short write occurred.